In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df = sns.load_dataset('iris')
df

sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]

***CHECKING MISSING VALUES***

In [3]:
df.isnull().sum()

sepal_length    0
sepal_width     0
petal_length    0
petal_width     0
species         0
dtype: int64

***DATA SEPARATIO***N

In [4]:
x = df.iloc[:,0:4]
x

sepal_length  sepal_width  petal_length  petal_width
0             5.1          3.5           1.4          0.2
1             4.9          3.0           1.4          0.2
2             4.7          3.2           1.3          0.2
3             4.6          3.1           1.5          0.2
4             5.0          3.6           1.4          0.2
..            ...          ...           ...          ...
145           6.7          3.0           5.2          2.3
146           6.3          2.5           5.0          1.9
147           6.5          3.0           5.2          2.0
148           6.2          3.4           5.4          2.3
149           5.9          3.0           5.1          1.8

[150 rows x 4 columns]

In [5]:
y =df.iloc[:,4]
y

0         setosa
1         setosa
2         setosa
3         setosa
4         setosa
         ...    
145    virginica
146    virginica
147    virginica
148    virginica
149    virginica
Name: species, Length: 150, dtype: object

In [7]:
df.describe

<bound method NDFrame.describe of      sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]>

In [8]:
df.shape

(150, 5)

***TRAIN TEST SPLIT***

In [9]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.2)
x_train

sepal_length  sepal_width  petal_length  petal_width
1             4.9          3.0           1.4          0.2
52            6.9          3.1           4.9          1.5
139           6.9          3.1           5.4          2.1
101           5.8          2.7           5.1          1.9
67            5.8          2.7           4.1          1.0
..            ...          ...           ...          ...
96            5.7          2.9           4.2          1.3
64            5.6          2.9           3.6          1.3
44            5.1          3.8           1.9          0.4
105           7.6          3.0           6.6          2.1
148           6.2          3.4           5.4          2.3

[120 rows x 4 columns]

KNN ***CLASSIFIER***

In [10]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score

In [11]:
knn = KNeighborsClassifier(n_neighbors = 15)
knn.fit(x_train,y_train)
y_predict = knn.predict(x_test)
y_predict

array(['setosa', 'versicolor', 'versicolor', 'setosa', 'versicolor',
       'setosa', 'versicolor', 'setosa', 'setosa', 'virginica',
       'virginica', 'versicolor', 'virginica', 'virginica', 'versicolor',
       'setosa', 'versicolor', 'virginica', 'setosa', 'versicolor',
       'setosa', 'versicolor', 'virginica', 'versicolor', 'virginica',
       'versicolor', 'versicolor', 'virginica', 'versicolor', 'virginica'],
      dtype=object)

***ACCURACY***

In [12]:
accuracy_score(y_test,y_predict)

0.9666666666666667

***CONFUSION MATRIX***

In [13]:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test,y_predict)
cm

array([[ 8,  0,  0],
       [ 0, 12,  0],
       [ 0,  1,  9]])

***CLASSIFICATION REPORT***

In [14]:
from sklearn.metrics import classification_report
print(classification_report(y_test,y_predict))

              precision    recall  f1-score   support

      setosa       1.00      1.00      1.00         8
  versicolor       0.92      1.00      0.96        12
   virginica       1.00      0.90      0.95        10

    accuracy                           0.97        30
   macro avg       0.97      0.97      0.97        30
weighted avg       0.97      0.97      0.97        30



***CONCLUSION***

**I have performed Iris Flower Classification using KNN Classifier.** 

**Accuracy of KNN Classifier is 96.67%.** 